File: motorph_payroll.py
# MotorPH Payroll System 

import json

DATA_FILE = "employees.json"

class Employee:
    def __init__(self, emp_id, name, position, rate_per_hour):
        self.emp_id = emp_id
        self.name = name
        self.position = position
        self.rate_per_hour = rate_per_hour

    def to_dict(self):
        return {
            "emp_id": self.emp_id,
            "name": self.name,
            "position": self.position,
            "rate_per_hour": self.rate_per_hour
        }

class PayrollSystem:
    def __init__(self):
        self.employees = {}
        self.load_data()

    def load_data(self):
        try:
            with open(DATA_FILE, "r") as f:
                data = json.load(f)
                for emp_id, emp in data.items():
                    self.employees[emp_id] = Employee(
                        emp["emp_id"],
                        emp["name"],
                        emp["position"],
                        emp["rate_per_hour"]
                    )
        except:
            pass

    def save_data(self):
        with open(DATA_FILE, "w") as f:
            json.dump({eid: emp.to_dict() for eid, emp in self.employees.items()}, f)

    def add_employee(self):
        emp_id = input("Enter Employee ID: ")
        name = input("Enter Name: ")
        position = input("Enter Position: ")
        rate = float(input("Enter Hourly Rate: "))

        self.employees[emp_id] = Employee(emp_id, name, position, rate)
        self.save_data()
        print("Employee added!\n")

    def view_employees(self):
        if not self.employees:
            print("No employees.\n")
            return

        for emp in self.employees.values():
            print(emp.emp_id, emp.name, emp.position, emp.rate_per_hour)
        print()

    def compute_salary(self):
        emp_id = input("Enter Employee ID: ")
        if emp_id not in self.employees:
            print("Not found.\n")
            return

        hours = float(input("Hours worked: "))
        emp = self.employees[emp_id]

        gross = hours * emp.rate_per_hour
        tax = gross * 0.1
        net = gross - tax

        print("\nPayslip")
        print("Name:", emp.name)
        print("Gross:", gross)
        print("Tax:", tax)
        print("Net:", net, "\n")

    def menu(self):
        while True:
            print("1. Add Employee")
            print("2. View Employees")
            print("3. Compute Salary")
            print("4. Exit")

            choice = input("Choice: ")

            if choice == "1":
                self.add_employee()
            elif choice == "2":
                self.view_employees()
            elif choice == "3":
                self.compute_salary()
            elif choice == "4":
                break

if __name__ == "__main__":
    system = PayrollSystem()
    system.menu()

File: MotorPHPayroll.java
import java.util.*;

class Employee {
    String id;
    String name;
    double rate;

    Employee(String id, String name, double rate) {
        this.id = id;
        this.name = name;
        this.rate = rate;
    }
}

public class Main {
    static HashMap<String, Employee> employees = new HashMap<>();
    static Scanner sc = new Scanner(System.in);

    public static void addEmployee() {
        System.out.print("Enter Employee ID: ");
        String id = sc.nextLine();

        System.out.print("Enter Name: ");
        String name = sc.nextLine();

        System.out.print("Enter Hourly Rate: ");
        double rate = sc.nextDouble();
        sc.nextLine();

        employees.put(id, new Employee(id, name, rate));
        System.out.println("Employee added!");
    }

    public static void viewEmployees() {
        if (employees.isEmpty()) {
            System.out.println("No employees found.");
            return;
        }

        for (Employee e : employees.values()) {
            System.out.println("ID: " + e.id + ", Name: " + e.name + ", Rate: " + e.rate);
        }
    }

    public static void computeSalary() {
        System.out.print("Enter Employee ID: ");
        String id = sc.nextLine();

        if (!employees.containsKey(id)) {
            System.out.println("Employee not found.");
            return;
        }

        System.out.print("Enter hours worked: ");
        double hours = sc.nextDouble();
        sc.nextLine();

        Employee e = employees.get(id);
        double salary = hours * e.rate;

        System.out.println("Salary of " + e.name + ": " + salary);
    }

    public static void main(String[] args) {
        while (true) {
            System.out.println("\n1. Add Employee");
            System.out.println("2. View Employees");
            System.out.println("3. Compute Salary");
            System.out.println("4. Exit");

            System.out.print("Enter choice: ");
            String choice = sc.nextLine();

            if (choice.equals("1")) {
                addEmployee();
            } else if (choice.equals("2")) {
                viewEmployees();
            } else if (choice.equals("3")) {
                computeSalary();
            } else if (choice.equals("4")) {
                break;
            } else {
                System.out.println("Invalid choice");
            }
        }
    }
}